packages/services/hmi-server/src/main/java/software/uncharted/terarium/hmiserver/resources/TransformResource.java

```diff
@@ -32,6 +37,42 @@ public Response mathML2ACSet(List<String> list) {
 		return skemaProxy.convertMathML2ACSet(list);
 	}
 
+	@POST
+	@Path("/mathml-to-amr/{framework}")
+	@APIResponses({
+		@APIResponse(responseCode = "500", description = "An error occurred retrieving the AMR"),
+		@APIResponse(responseCode = "400", description = "Query must contain a valid MathML and a Model framework")
+	})
+	public Response mathML2AMR(
+		final List<String> mathml,
+		@PathParam("framework") final String framework
+	) {
+
+		// Check if we have a framework and a MathML
+		if (framework == null || mathml == null || mathml.isEmpty()) {
+			return Response.status(Response.Status.BAD_REQUEST).build();
+		}
+
+		try {
+			// Create Request Body
+			// https://skema-rs.terarium.ai/docs/#/skema%3A%3Aservices%3A%3Amathml/get_amr
+			ObjectMapper mapper = new ObjectMapper();
+			ObjectNode request = mapper.createObjectNode();
+			request.put("model", framework);
+			request.put("mathml", mapper.valueToTree(mathml));
+			Response response = skemaProxy.convertMathML2AMR(request);
+
+			if (response.getStatus() != Response.Status.OK.ordinal()) {
+				Response.status(Response.Status.BAD_REQUEST).build();
+			}
```

Comment:
If you did the input validation above, then this is arguably a sever error and not a Bad-Request.

Reply:
I agree, but for now, because we keep sending bad MathML to the service, and they have poor error handling, I decided to say it was a bad request and precised in the error message above that the MathML needs to be valid and mathematically sound.